### PR TITLE
[FIRRTL] Add edge attribute to LTL clock intrinsic

### PR DESCRIPTION
**Pull request by Clo91eaf**
Add edge attribute to LTL clock intrinsic in FIRRTL dialect, which should be aligned to [ltl dialect](https://circt.llvm.org/docs/Dialects/LTL/#ltlclock-circtltlclockop)

Assisted-by: Codex:gpt-5.5

### CODE REVIEW

**Comment by Clo91eaf** [comment]
@seldridge cc

**Comment by Clo91eaf** [comment]
@seldridge cc

File: test/Dialect/FIRRTL/lower-intrinsics.mlir [hunk 1]
```diff
@@ -80,8 +80,10 @@ firrtl.circuit "Foo" {
     // CHECK-NEXT: firrtl.int.ltl.past %in0, 42, %clk :
     firrtl.int.generic "circt_ltl_past" <delay: i64 = 42> %in0, %clk : (!firrtl.uint<1>, !firrtl.clock) -> !firrtl.uint<1>
 
-    // CHECK-NEXT: firrtl.int.ltl.clock %in0, %clk :
+    // CHECK-NEXT: firrtl.int.ltl.clock %in0, posedge %clk :
     firrtl.int.generic "circt_ltl_clock"  %in0, %clk : (!firrtl.uint<1>, !firrtl.clock) -> !firrtl.uint<1>
+    // CHECK-NEXT: firrtl.int.ltl.clock %in0, negedge %clk :
+    firrtl.int.generic "circt_ltl_clock" <edge: none = "negedge"> %in0, %clk : (!firrtl.uint<1>, !firrtl.clock) -> !firrtl.uint<1>
```
Comment (seldridge) [inline test/Dialect/FIRRTL/lower-intrinsics.mlir:90]: Please include both a lowering test of `edge = "posedge"` (explicit posedge) and `edge = "edge"`.

File: test/Dialect/FIRRTL/expand-whens.mlir
Comment (seldridge) [inline test/Dialect/FIRRTL/expand-whens.mlir:1]: I would _not_ change this test to be all `negedge`. Can you instead add a separate test section at the bottom for the `negedge` testing?

**Review by seldridge — APPROVED** [review]
Generally looks good.  Only a couple of nits.  Please address these before landing.

**Comment by Clo91eaf** [comment]
@seldridge cc

**Review by seldridge — APPROVED** [review]

**Comment by Clo91eaf** [comment]
@seldridge hi! can you merge it?
